City-of-Helsinki/youth-membership-ui
# YM-379 | View own information tests

## Description
- Added `e2e` - tests that navigate to "Profile information" page and checks some of the entered values (if needs be this can be extended to cover all information user entered. Relating to back-end separation, would this actually be a good idea?).

## Context
Tests are always good 😄 

User birth year values are no longer hardcoded. This will prevent tests from crashing in the distant future.

I also created `.testcaferc.json` file. This was necessary to alter the order in which testcafe runs the tests, as one would expect, it uses alphabetical order as default. Moving file order to different file should also keep `package.json` script section little cleaner. I decided to leave browser information inside `browser-test` scripts for now.

[YM-379](https://helsinkisolutionoffice.atlassian.net/browse/YM-379)

## How Has This Been Tested?
`yarn browser-test` -> all test pass.

## Manual Testing Instructions for Reviewers

1) Add values to `.env` file (if you don't have these you can contact me):
- `REACT_APP_TESTING_USERNAME`
- `REACT_APP_TESTING_PASSWORD`
2) Login to [profile-admin](https://profiili-api.test.kuva.hel.ninja/admin) and make sure "Uno User" has no profile
3) Start local application `yarn start`
4) Run `yarn browser-test`
5) Delete the created profile to avoid crashing next time tests are run.
